src/main/java/org/opensearch/security/support/HostAndCidrMatcher.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.support;
+
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import inet.ipaddr.IPAddressString;
+
+/**
+ * A utility class that performs matching of IP addresses against hostname patterns and CIDR ranges.
+ * This matcher supports both wildcard hostname patterns (e.g., *.example.com) and CIDR notation (e.g., 192.168.1.0/24).
+ */
+public class HostAndCidrMatcher {
+    private static final String IP_HOSTNAME = "ip-hostname";
```

Comment:
Nit: Would it be possible to use an enum for `IP_HOSTNAME` and `IO_HOSTNAME_LOOKUP`?

Reply:
ack, done.